### PR TITLE
Rename semi join to left semi join

### PR DESCRIPTION
**Pull request by majetideepak**
Semi join implementation is a left semi join implementation and 
is renamed accordingly.

### CODE REVIEW

File: velox/core/PlanNode.cpp [hunk 1]
```diff
@@ -332,9 +332,6 @@ AbstractJoinNode::AbstractJoinNode(
       leftKeys_.size(),
       rightKeys_.size(),
       "JoinNode requires same number of join keys on left and right sides");
-  if (isSemiJoin() || isAntiJoin()) {
-    VELOX_CHECK_NULL(filter, "Semi and anti join does not support filter");
```
Comment (majetideepak) [inline velox/core/PlanNode.cpp:336]: all joins seem to support a filter.
This is dead code, to begin with since the member `filter_` has the filters.

**Review by mbasmanova — APPROVED** [review]
Thanks. Should we update https://facebookincubator.github.io/velox/develop/operators.html as well?

**Comment by majetideepak** [comment]
> Thanks. Should we update https://facebookincubator.github.io/velox/develop/operators.html as well?

@mbasmanova fixed. Good find!

**Review by aditi-pandit — APPROVED** [review]
Thanks Deepak. Makes the code more readable.

**Pull request closed without merging**